### ISSUE TITLE
Fix S1172 FP: Issue is raised when parameter is used in tuple creation

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3255]
As reported on this [community thread](https://community.sonarsource.com/t/c-s1172-parameters-not-recognized-in-value-tuples/22011/6)

### Description

```C#
public string UsedInTuple(string value) // Noncompliant FP
{
    var x = (value, 7);
    return x.value;
}
```

### Related information

* SonarC# 8.5

**Comment by pavel-mikula-sonarsource** [comment]
This is not reproducible any more on current master. There was probably a [tooling issue](https://github.com/SonarSource/sonar-dotnet/commit/81fc740d5b4ce92638ec2f1052e8a61c72cfc93d) causing this to be false-reproduced.

**Comment by pavel-mikula-sonarsource** [comment]
This is still reproducible when DLL is referenced directly from a project.
But it's not reproducible inside a test (see [this commit](https://github.com/SonarSource/sonar-dotnet/commit/81fc740d5b4ce92638ec2f1052e8a61c72cfc93d)
```xml
  <ItemGroup>
    <Analyzer Include="c:\Projects\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\\Google.Protobuf.dll" />
    <Analyzer Include="c:\Projects\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\\SonarAnalyzer.CFG.dll" />
    <Analyzer Include="c:\Projects\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\\SonarAnalyzer.CSharp.dll" />
    <Analyzer Include="c:\Projects\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\\SonarAnalyzer.dll" />
  </ItemGroup>
```

**Comment by markusschaber** [comment]
Just stumbled across this in our code. Any estimate when a fix for this will be released?

**Comment by pavel-mikula-sonarsource** [comment]
Hi @markusschaber, we don't have any specific estimation. Rough guess is that we'll try to fix this until January 2021.
